### PR TITLE
Added DSA section in courses for Tamil Language and added a playlist on DSA in Tamil

### PR DESCRIPTION
**Pull request by yogan-gopi**
## What does this PR do?
I've added a new section **DSA(Data Structures & Algorithms)** in courses for **Tamil** language and also added a playlist on DSA in TAmil

### Why is this valuable (or not)?
It is valuable, as DSA is really important in programming.

### How do we know it's really free?
Yes, it is free.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by yogan-gopi** [comment]
Closes #7776

**Comment by eshellman** [comment]
use 'Data Structures and Algorithms', not DSA

**Comment by yogan-gopi** [comment]
> use 'Data Structures and Algorithms', not DSA

Done!

File: courses/free-courses-ta.md [hunk 1]
```diff
@@ -35,6 +36,12 @@
 * [C++ Programming in Tamil](https://youtube.com/playlist?list=PLWbtDrDnmTHBPD-Dt5BJi7iP11x6UvgU0) - CS in Tamil
 
 
+### <a id="dsa"></a>Data Structures & Algorithms
```
Comment (eshellman) [inline courses/free-courses-ta.md:39]:
```suggestion
### <a id="dsa"></a>Data Structures and Algorithms
```

**Comment by yogan-gopi** [comment]
Done!

**Pull request merged**